requirements.txt

```diff
@@ -7,6 +7,7 @@ onnx==1.15.0
 huggingface-hub==0.22.2
 transformers==4.40.0
 sentencepiece==0.2.0
+datasets==3.0.0
```

Comment:
We should steer towards putting requirements in the subproject requirements files instead of this top level file, especially if this is a test-only requirement

* Existing requirements file specific to sharktank: https://github.com/nod-ai/SHARK-Platform/blob/main/sharktank/requirements.txt
* Test requirements file for shortfin: https://github.com/nod-ai/SHARK-Platform/blob/main/shortfin/requirements-tests.txt